src/main/java/pluto/command/UpdateStatusCommand.java

```diff
@@ -0,0 +1,38 @@
+package pluto.command;
+
+import pluto.PlutoException;
+import pluto.Storage;
+import pluto.TaskList;
+import pluto.Ui;
+
+import java.io.IOException;
+
+public class UpdateStatusCommand extends Command {
+
+    private int idx;
+    private boolean isDone;
+
+    public UpdateStatusCommand(int idx, boolean isDone) {
+        this.idx = idx;
+        this.isDone = isDone;
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws PlutoException {
+        tasks.markTask(idx, isDone);
+        try {
+            storage.rewriteFile(tasks);
+            StringBuilder markMessage = new StringBuilder();
+            if (isDone) {
+                markMessage.append("\tNice! I've marked this task as done:\n");
+            } else {
+                markMessage.append("\tOK, I've marked this task as not done yet:\n");
+            }
+            markMessage.append("\t\t" + tasks.getTask(idx).toString());
+            ui.print(markMessage);
+        } catch (IOException e) {
+            tasks.markTask(idx, !isDone);
+            throw new PlutoException("\tOOPS!!! Couldn't update task status. Try again!");
+        }
+    }
```

Comment:
I like that you combined both Mark and Unmark command into a single command. It is a very efficient use of code. Really clean.

Unfortunately, this code block still shares the issues I had mentioned in DeleteCommand. Refer to my comments there for more details.